### PR TITLE
add the sigmoid, Sigmoid for the api 2.0

### PR DESCRIPTION
**Pull request by wawltor**
### PR types
 Others

### PR changes
 APIs

### Describe
add the sigmoid, Sigmoid for the api 2.0

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/test_nn_sigmoid_op.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+                      feed={'x': self.x},
+                      fetch_list=[y, y.grad_name, x.grad_name])
+        self.assertTrue(np.allclose(out[0], self.y))
+        self.assertTrue(np.allclose(out[2], self.ref_backward(self.y, out[1])))
+        self.assertTrue(y.name.startswith("api_sigmoid"))
+
+    def check_dynamic_api(self, place):
+        paddle.disable_static(place)
+        x = paddle.to_variable(self.x)
+        mysigmoid = nn.Sigmoid()
```
Comment (T8T9) [inline python/paddle/fluid/tests/unittests/test_nn_sigmoid_op.py:63]: 动态图需不需要检查反向？

File: python/paddle/nn/layer/activation.py [hunk 2]
```diff
@@ -250,47 +250,43 @@ def forward(self, x):
...
-          exe.run(fluid.default_startup_program())
+          import paddle
+          import paddle.fluid as fluid
+
+          paddle.disable_static()
           input_data = np.array([1.0, 2.0, 3.0, 4.0]).astype('float32')
-          output_data = exe.run(feed={"input": input_data},
-                                fetch_list=[output])
-          print(output_data) # [0.7310586, 0.880797, 0.95257413, 0.98201376]
+          sigmoid = paddle.nn.Sigmoid()
```
Comment (T8T9) [inline python/paddle/nn/layer/activation.py:278]: 感觉 m = paddle.nn.Sigmoid() 好一些，和torch 中的一致

File: python/paddle/nn/layer/activation.py [hunk 3]
```diff
@@ -250,47 +250,43 @@ def forward(self, x):
...
-    def __init__(self, inplace=False):
+    def __init__(self, name=None):
         super(Sigmoid, self).__init__()
-        self._inplace = inplace
+        self.name = name
 
-    def forward(self, input):
-        return functional.sigmoid(input, self._inplace)
+    def forward(self, x):
+        return functional.sigmoid(x, self.name)
```
Comment (T8T9) [inline python/paddle/nn/layer/activation.py:289]: 当时好像有说用`import functional as F`来着？

Comment (wawltor) [reply python/paddle/nn/layer/activation.py]: fixed it

File: python/paddle/nn/layer/activation.py (hunk 2 above)
Comment (wawltor) [reply python/paddle/nn/layer/activation.py]: fixed it

File: python/paddle/fluid/layers/ops.py [hunk 4]
```diff
@@ -58,9 +61,18 @@
...
 for _OP in set(__activations_noattr__):
-    globals()[_OP] = generate_activation_fn(_OP)
+    func = generate_activation_fn(_OP)
+    func = deprecated(
+        since="2.0.0", update_to="paddle.nn.functional.%s" % (_OP))(func)
+    globals()[_OP] = func
+
+for _OP in set(__unary_func__):
+    func = generate_activation_fn(_OP)
+    func = deprecated(since="2.0.0", update_to="paddle.%s" % (_OP))(func)
```
[the remark this answers is not in the document]
Comment (wawltor) [reply python/paddle/fluid/layers/ops.py:74]: 目前是保持一致的

File: python/paddle/fluid/layers/ops.py [hunk 5]
```diff
@@ -23,26 +23,29 @@
 __activations_noattr__ = [
     'sigmoid',
     'logsigmoid',
+    'tanh_shrink',
```
[the remark this answers is not in the document]
Comment (wawltor) [reply python/paddle/fluid/layers/ops.py:26]: 目前nn下面和layers的是保持一致的

File: python/paddle/fluid/tests/unittests/test_nn_sigmoid_op.py (hunk 1 above)
Comment (wawltor) [reply python/paddle/fluid/tests/unittests/test_nn_sigmoid_op.py:60]: 沟通后先不增加

**Comment by wawltor** [comment]
图片
图片
图片
图片
图片
图片

File: python/paddle/nn/layer/activation.py [hunk 6]
```diff
@@ -245,52 +245,48 @@ def __init__(self, alpha=1e-2, name=None):
...
-          import paddle.fluid as fluid
-          import paddle.nn as nn
           import numpy as np
-          input = fluid.data(name="input", shape=[None, 4])
-          output = nn.Sigmoid()(input)
-          place = fluid.CPUPlace()
-          exe = fluid.Executor(place)
-          exe.run(fluid.default_startup_program())
+          import paddle
+          import paddle.fluid as fluid
```
Comment (TCChenlong) [inline python/paddle/nn/layer/activation.py]: 这行不应该出现~

**Review by TCChenlong — DISMISSED** [review]
LGTM

**Comment by T8T9** [comment]
LGTM

**Review by FlyingQianMM — APPROVED** [review]
LGTM
